oak-auth-external/src/main/java/org/apache/jackrabbit/oak/spi/security/authentication/external/impl/principal/DynamicGroupUtil.java

```diff
@@ -65,7 +66,7 @@ static boolean isMemberProperty(@NotNull PropertyState propertyState) {
 
     @NotNull
     static Tree getTree(@NotNull Authorizable authorizable, @NotNull Root root) throws RepositoryException {
-        return root.getTree(authorizable.getPath());
+        return (authorizable instanceof TreeAware) ? ((TreeAware) authorizable).getTree() : root.getTree(authorizable.getPath());
```

Comment:
I see this expression quite often in this patch, but spread across multiple modules. Do you see a sensible location where we could centralize this code and just reference it? Just like the ```Utils.getTree()``` method below.

Reply:
hi @joerghoh , good point. i actually was tempted to add it to `UserUtil` in _oak-security-spi_ the reason why i decided against it: the path of a User/Group is a JCR path. so to it 'right' it in a generic way the call would also need a `NamePathMapper` to convert the JCR path to an Oak path as it is required for the` Root.getTree` call. But for all usages (except maybe for one) i know that the user/group has. been retrieved using Oak API which doesn't have any namespace-remapping in place and thus decided to live with the duplication instead.